packages/expo-notifications/plugin/src/__tests__/withNotificationsiOS-test.ts

```diff
@@ -0,0 +1,48 @@
+import { IOSConfig } from '@expo/config-plugins';
+import { fs, vol } from 'memfs';
+import * as path from 'path';
+
+import { setNotificationSounds } from '../withNotificationsIOS';
+import { getDirFromFS } from './withNotificationsAndroid-test';
+
+jest.mock('fs');
+
+const fsReal = jest.requireActual('fs') as typeof fs;
+
+const LIST_OF_GENERATED_FILES = [
+  'assets/notificationSound.wav',
+  'ios/testproject/notificationSound.wav',
+  'ios/testproject.xcodeproj/project.pbxproj',
+  'ios/testproject/AppDelegate.m',
+];
+
+const soundPath = path.resolve(__dirname, './fixtures/cat.wav');
+
+const projectRoot = '/app';
+
+describe('iOS notifications configuration', () => {
+  beforeAll(async () => {
+    const sound = fsReal.readFileSync(soundPath);
+    vol.fromJSON({ 'ios/testproject/AppDelegate.m': '' }, projectRoot);
+    vol.mkdirpSync('/app/assets');
+    vol.mkdirpSync('/app/ios/testproject.xcodeproj/');
+    vol.writeFileSync('/app/assets/notificationSound.wav', sound);
+    vol.writeFileSync(
+      '/app/ios/testproject.xcodeproj/project.pbxproj',
+      fsReal.readFileSync(path.join(__dirname, 'fixtures/project.pbxproj'), 'utf-8')
+    );
+
+    const project = IOSConfig.XcodeUtils.getPbxproj(projectRoot);
+    setNotificationSounds(['/app/assets/notificationSound.wav'], { project, projectRoot });
```

Comment:
Code that you're testing should live in the `it` statement as that'll handle errors in a more clear fashion.